docs/intro/tutorial02.txt

```diff
@@ -463,7 +463,7 @@ Let's also add a custom method to this model:
     class Question(models.Model):
         # ...
         def was_published_recently(self):
-            return self.pub_date >= timezone.now() - datetime.timedelta(days=1)
+            return self.pub_date >= timezone.now().date() - datetime.timedelta(days=1)
```

Comment:
It seems to me that your change introduces a `TypeError` and that the tutorial is fine as-is:

1. `pub_date` is a `DateTimeField` which is represented as a `datetime.datetime` in Python
2. `timezone.now()` returns a `datetime.datetime`
3. A `datetime.timedelta` can be subtracted from a `datetime.datetime` instance

This is what I'm getting:

```python
>>> import datetime
>>> from django.utils import timezone
>>>
>>> datetime.datetime.now() >= timezone.now() - datetime.timedelta(days=1)
True
>>> datetime.datetime.now() >= timezone.now().date() - datetime.timedelta(days=1)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: can't compare datetime.datetime to datetime.date
```

Are you seeing something different?

Reply:
That's funny, I also have an error on the first one...
<img width="723" alt="Screenshot 2024-04-15 at 17 37 45" src="https://github.com/django/django/assets/79864209/22626ce2-26ee-4211-9f59-8666553cdeb9">

Since we're at it, here's the output I have on the console. I get true with the `.date()`, but the error otherwise.
<img width="864" alt="Screenshot 2024-04-15 at 17 41 26" src="https://github.com/django/django/assets/79864209/858c6914-8f55-4bce-a757-d67a33e27f56">

Reply:
I have `USE_TZ` set to `False` for simplicity. I think that's why you got the first error.

I'll run through the tutorial soon and compare.

Reply:
I can't reproduce your error from the tutorial instructions. What does `print(foo.pub_date)` give you?

Reply:
I don't have `USE_TZ` defined.

<img width="1296" alt="Screenshot 2024-04-15 at 18 25 50" src="https://github.com/django/django/assets/79864209/d0e7ea8e-830e-415e-8c9b-bfe1ccfcd813">

Reply:
I followed the tutorial by the book, IIRC.

Here's my models.py file:
```
import datetime

from django.db import models
from django.utils import timezone


class Question(models.Model):
    question_text = models.CharField(max_length=200)
    pub_date = models.DateField("date published")

    def __str__(self):
        return self.question_text

    def was_published_recently(self):
        return self.pub_date >= timezone.now() - datetime.timedelta(days=1)


class Choice(models.Model):
    question = models.ForeignKey(Question, on_delete=models.CASCADE)
    choice_text = models.CharField(max_length=200)
    votes = models.IntegerField(default=0)

    def __str__(self):
        return self.choice_text
```

Reply:
I have Django version 5.0.4, working on a Mac M1. But then again, I'm in a virtual env, so I wonder if this would really make a difference.

Reply:
Thank you for posting your code! You defined the `pub_date` field as a `DateField` instead of a `DateTimeField`. That's the problem.

https://docs.djangoproject.com/en/5.0/intro/tutorial02/#creating-models

Reply:
Also, `USE_TZ` is a Django setting. No need to think about that right now. The confusion was my fault!

Reply:
@jsantos42 `USE_TZ` is a Django setting, not an environment variable.

Reply:
Thank you guys! My bad, sorry for wasting your time, I was just trying to help 🙈
Cheers!

Reply:
No worries! Glad to help.